crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -212,19 +212,23 @@ impl<TX: DbTx> DatabaseProvider<TX> {
         segment: SnapshotSegment,
         block_range: Range<u64>,
```

Comment:
not only a block range now, right? Also can be a tx range, same as in `get_with_snapshot`